pkg/controller/installlogmonitor/installlogmonitor_controller.go

```diff
@@ -0,0 +1,280 @@
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+)
+
+const (
+	controllerName      = "installlogmonitor"
+	processedAnnotation = "hive.openshift.io/install-log-processed"
+	regexConfigMapName  = "install-log-regexes"
+	hiveNamespace       = "hive"
+	unknownReason       = "UnknownError"
+	unknownMessage      = "Cluster install failed but no known errors found in logs"
+	successReason       = "ClusterInstalled"
+	successMessage      = "Cluster install completed successfully"
+)
+
+var (
+	metricInstallErrors = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "hive_install_errors",
+		Help: "Counter incremented every time we observe certain errors strings in install logs.",
+	},
+		[]string{"cluster_type", "reason"},
+	)
+)
+
+func init() {
+	metrics.Registry.MustRegister(metricInstallErrors)
+}
+
+// Add creates a new InstallLogMonitro Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileInstallLog{Client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("installlogmonitor-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to ConfigMap
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileInstallLog{}
+
+// ReconcileInstallLog reconciles an install log ConfigMap object uploaded by the hive installmanager pod.
+type ReconcileInstallLog struct {
+	client.Client
+	scheme *runtime.Scheme
+}
+
+func (r *ReconcileInstallLog) isHiveInstallLog(cm *corev1.ConfigMap) (isInstallLog bool, needsMigration bool) {
+	if _, ok := cm.Labels[hivev1.HiveInstallLogLabel]; ok {
+		return true, false
+	}
+
+	// If it looks like one of our install logs that predates install manager images, return true
+	// and indicate it needs migration.
+	if _, ok := cm.Data["log"]; ok && strings.HasSuffix(cm.Name, "-install-log") {
+		return true, true
+	}
+
+	return false, false
+}
+
+// migrateInstallLog is a temporary function that will apply the new labels to pre-existing install log configmaps.
+func (r *ReconcileInstallLog) migrateInstallLog(cm *corev1.ConfigMap) error {
+	if cm.Labels == nil {
+		cm.Labels = map[string]string{}
+	}
+
+	cm.Labels[hivev1.HiveInstallLogLabel] = "true"
+
+	// We must guess at the cluster deployment name:
+	clusterDeploymentName := cm.Name[:len(cm.Name)-len("-install-log")]
```

Comment:
would it be safer to look at the owner reference?

Reply:
Good idea, updated.